### PR TITLE
feat: configure basic authentication for publishing

### PR DESCRIPTION
**Pull request by kov117**
### Checklist

- [x] I've updated both the `azurerm_linux_function_app` and `azurerm_windows_function_app` resources.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -161,6 +168,12 @@ resource "azurerm_windows_function_app" "this" {
   app_settings                = var.app_settings
   functions_extension_version = var.functions_extension_version
 
+  builtin_logging_enabled                        = var.builtin_logging_enabled
+  client_certificate_mode                        = var.client_certificate_mode
+  ftp_publish_basic_authentication_enabled       = var.ftp_publish_basic_authentication_enabled
+  webdeploy_publish_basic_authentication_enabled = var.webdeploy_publish_basic_authentication_enabled
+
+
```
Comment (sefornes) [inline main.tf:176]: Remove the redundant space?

**Comment by hknutsen** [comment]
Variable names, types and defaults look good according to best practices 👍

**Comment by hknutsen** [comment]
... actually, I'd argue that default values should be set to `false` to increase security 👀 https://learn.microsoft.com/en-us/azure/app-service/configure-basic-auth-disable

**Comment by kov117** [comment]
Ah i see, thanks for letting me know 🙌🏼 will change them to false

**Comment by hknutsen** [comment]
> ... actually, I'd argue that default values should be set to `false` to increase security 👀 https://learn.microsoft.com/en-us/azure/app-service/configure-basic-auth-disable

@equinor/terraform-baseline-maintainers Agree?

File: variables.tf [hunk 2]
```diff
@@ -216,6 +216,18 @@ variable "identity_ids" {
   default     = []
 }
 
+variable "ftp_publish_basic_authentication_enabled" {
+  description = "Should basic (username and password) authentication be enabled for the FPT client?"
```
Comment (hknutsen) [inline variables.tf:220]: I did a typo...

```suggestion
  description = "Should basic (username and password) authentication be enabled for the FTP client?"
```

**Review by hknutsen — DISMISSED** [review]

File: variables.tf (hunk 2 above)
Comment (kov117) [reply variables.tf:220]: no worries, i missed it aswell 😆

**Comment by helenakallekleiv** [comment]
> > ... actually, I'd argue that default values should be set to `false` to increase security 👀 https://learn.microsoft.com/en-us/azure/app-service/configure-basic-auth-disable
> 
> @equinor/terraform-baseline-maintainers Agree?

Agreed.

**Comment by kov117** [comment]
Ran a new test and it passed 👍🏼

**Review by helenakallekleiv — APPROVED** [review]

**Pull request merged**